include/QBDI/Memory.h

```diff
@@ -43,34 +44,74 @@ typedef enum {
     _QBDI_EI(PF_EXEC) = 4    /*!< Execution access */
 } Permission;
 
+#ifdef __cplusplus
 _QBDI_ENABLE_BITMASK_OPERATORS(Permission)
+#endif
 
 /*! Map of a memory area (region).
  */
-class MemoryMap {
-
-public:
+#ifdef __cplusplus
```

Comment:
It's cleaner to split C++ / C API into two files:
- Memory.h for the C API
- Memory.hpp for the C++ API

Also make sure to split definition from implementation